examples/view-integration/external-views/src/containerCode.ts

```diff
@@ -32,26 +34,31 @@ const diceRollerId = "dice-roller";
 export class DiceRollerContainerRuntimeFactory extends ModelContainerRuntimeFactory<IDiceRollerAppModel> {
 	constructor() {
 		super(
-			new Map([DiceRollerInstantiationFactory.registryEntry]), // registryEntries
+			new Map([[diceRollerRegistryKey, Promise.resolve(diceRollerFactory)]]), // registryEntries
 		);
 	}
 
 	/**
 	 * {@inheritDoc ModelContainerRuntimeFactory.containerInitializingFirstTime}
 	 */
 	protected async containerInitializingFirstTime(runtime: IContainerRuntime) {
-		const diceRoller = await runtime.createDataStore(DiceRollerInstantiationFactory.type);
+		const diceRoller = await runtime.createDataStore(diceRollerRegistryKey);
 		await diceRoller.trySetAlias(diceRollerId);
 	}
 
 	/**
 	 * {@inheritDoc ModelContainerRuntimeFactory.createModel}
 	 */
 	protected async createModel(runtime: IContainerRuntime, container: IContainer) {
-		const diceRoller = await requestFluidObject<IDiceRoller>(
-			await runtime.getRootDataStore(diceRollerId),
-			"",
-		);
+		// If runtime.getRootDataStoreChannel was public we could just call that.
+		const diceRollerRuntime = (await runtime.getRootDataStore(
+			diceRollerId,
+		)) as IFluidDataStoreChannel;
+		const diceRollerEntryPoint = diceRollerRuntime.entryPoint;
+		if (diceRollerEntryPoint === undefined) {
+			throw new Error("EntryPoint was not defined");
+		}
+		const diceRoller = (await diceRollerEntryPoint.get()) as IDiceRoller;
```

Comment:
This side of things is kind-of optional for the core investigation of this PR - I mostly wanted to play with and better understand the new entryPoint pattern since it relates to initialization.  I think a couple things could improve it:
* If runtime.getRootDataStore() would return an IFluidDataStoreChannel straightaway [instead of an IFluidRouter](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/containerRuntime.ts#L2124) (maybe in plan once request() goes away?) then the first call becomes cleaner
* If [entryPoint was non-optional](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/datastore/src/dataStoreRuntime.ts#L133) we wouldn't need the undefined check (maybe also in plan for the entryPoint work once it's done?)
* If there was a way to strongly type the entryPoint somehow, maybe we could avoid the final cast - haven't really thought about this one though.

So with those changes it would at least end up something like:
```ts
		const diceRollerRuntime = await runtime.getRootDataStore(diceRollerId);
		const diceRoller = (await diceRollerRuntime.entryPoint.get()) as IDiceRoller;
```